src/instruction.rs

```diff
@@ -139,3 +141,122 @@ pub enum Instruction {
+            Instruction::Slliw(i) => write!(f, "slliw {} {} {}", i.rd(), i.rs1(), i.shamt()),
+            Instruction::Srliw(i) => write!(f, "srliw {} {} {}", i.rd(), i.rs1(), i.shamt()),
+            Instruction::Sraiw(i) => write!(f, "sraiw {} {} {}", i.rd(), i.rs1(), i.shamt()),
+            Instruction::Remuw(r) => write!(f, "remuw {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Flw(i) => write!(f, "flw {} {}({})", i.rd(), i.rs1(), i.imm()),
+            Instruction::Fsw(i) => write!(f, "fsw {} {}({})", i.rs2(), i.rs1(), i.imm()),
+            Instruction::Fmadds(r) => {
+                write!(f, "fmadds {} {} {} {}", r.rd(), r.rs1(), r.rs2(), r.rs3())
+            }
+            Instruction::Fmsubs(r) => {
+                write!(f, "fmsubs {} {} {} {}", r.rd(), r.rs1(), r.rs2(), r.rs3())
+            }
+            Instruction::Fnmsubs(r) => {
+                write!(f, "fnmsubs {} {} {} {}", r.rd(), r.rs1(), r.rs2(), r.rs3())
+            }
+            Instruction::Fnmadds(r) => {
+                write!(f, "fnmadds {} {} {} {}", r.rd(), r.rs1(), r.rs2(), r.rs3())
+            }
+            Instruction::Fadds(r) => write!(f, "fadds {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fsubs(r) => write!(f, "fsubs {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fmuls(r) => write!(f, "fmuls {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fdivs(r) => write!(f, "fdivs {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fsqrts(r) => write!(f, "fsqrts {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fsgnjs(r) => write!(f, "fsgnjs {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fsgnjns(r) => write!(f, "fsgnjns {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fsgnjxs(r) => write!(f, "fsgnjxs {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fmins(r) => write!(f, "fmins {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fmaxs(r) => write!(f, "fmaxs {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fcvtws(r) => write!(f, "fcvtws {} {}", r.rd(), r.rs1()),
+            Instruction::Fcvtwus(r) => write!(f, "fcvtwus {} {}", r.rd(), r.rs1()),
+            Instruction::Fmvxw(r) => write!(f, "fmvxw {} {}", r.rd(), r.rs1()),
+            Instruction::Feqs(r) => write!(f, "feqs {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Flts(r) => write!(f, "flts {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fles(r) => write!(f, "fles {} {} {}", r.rd(), r.rs1(), r.rs2()),
+            Instruction::Fclasss(r) => write!(f, "fclasss {} {}", r.rd(), r.rs1()),
+            Instruction::Fcvtsw(r) => write!(f, "fcvtsw {} {}", r.rd(), r.rs1()),
+            Instruction::Fcvtswu(r) => write!(f, "fcvtswu {} {}", r.rd(), r.rs1()),
+            Instruction::Fmvwx(r) => write!(f, "fmvwx {} {}", r.rd(), r.rs1()),
+            _ => write!(f, "unknown"),
```

Comment:
Are there any instructions missing? Would it be possible to switch this to an exhaustive match by panicking on Instruction::__Nonexhaustive?